### PR TITLE
NT-1171: Manage Pledge View Backing Info updates

### PR DESCRIPTION
**Pull request by JadeByfield89**
# 📲 What

Make changes to the Manage Pledge view to reflect the addition of the bonus support field, and some new UI changes.

# 🤔 Why

To begin adding support for backers who want to pledge a little extra to a project, and to make necessary UI changes that will be needed for add ons.

# 🛠 How

New UI changes that show bonus support, estimated delivery date and checkbox, and minor copy changes.

# 👀 See

Screen Shot 2020-06-16 at 7 50 28 PM

# 📋 QA
Users may view Add Ons they have backed on the web when viewing the View/Manage pledge screen 

In the list of numbers, a line called “Bonus” is added beneath shipping that denotes the amount of bonus support added by the backer

In the list of numbers, “Total amount” becomes “Total”

The Reward section is displayed with the title “Your pledge details”

The latest delivery date is displayed above the primary reward

Delivery dates are hidden from reward cards

The “Reward delivered” checkbox is displayed below the estimated delivery date

A section of helper copy/disclaimer copy is displayed above the primary reward

Design matches Figma 

Creators may view Add Ons a user has backed on the web when they view the pledge for a given backer

In the list of numbers, a line called “Bonus” is added beneath shipping that denotes the amount of bonus support added by the backer

In the list of numbers, “Total amount” becomes “Total”

The Reward section is displayed with the title “Pledge details”

The latest delivery date is displayed above the primary reward

Design matches Figma 

# Story 📖

https://kickstarter.atlassian.net/secure/RapidBoard.jspa?rapidView=9&modal=detail&selectedIssue=NT-1171

### CODE REVIEW

File: app/src/main/res/layout/fragment_backing_section_delivery_date_reminder.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    app:cardElevation="0dp">
+
+
+    <ImageView
+        android:id="@+id/icon_calendar"
+        android:background="@drawable/ic_calendar"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:contentDescription="@null"
```
[the remark this answers is not in the document]
Comment (JadeByfield89) [reply app/src/main/res/layout/fragment_backing_section_delivery_date_reminder.xml:18]: Done!

File: app/src/main/java/com/kickstarter/ui/fragments/BackingFragment.kt [hunk 2]
```diff
@@ -170,6 +183,53 @@ class BackingFragment: BaseFragment<BackingFragmentViewModel.ViewModel>(), Rewar
...
+        this.viewModel.environment.currentUser().observable()
+                .compose(bindToLifecycle())
+                .compose(Transformers.observeForUI())
+                .subscribe { user ->
+                    this.viewModel.outputs.projectDataAndReward().map { it.first }
+                            .compose(bindToLifecycle())
+                            .compose(Transformers.observeForUI())
+                            .subscribe { projectData ->
+
+                                if (ProjectUtils.userIsCreator(projectData.project(), user)) {
```
[the remark this answers is not in the document]
Comment (JadeByfield89) [reply app/src/main/java/com/kickstarter/ui/fragments/BackingFragment.kt:215]: Oh that's awesome, I had no idea about `combineLatest`. Thank you!

I'm a bit unclear about the need for these 2 additional observables
`receivedSectionIsGone(): Observable<Boolean>`
`deliveryDisclaimerSectionIsGone(): Observable<Boolean>`

Both of these pieces of UI are hidden if the user is the creator of the project, so would something like this suffice?

 ` this.viewModel.outputs.deliveryDisclaimerSectionIsGone()
                .compose(bindToLifecycle())
                .compose(Transformers.observeForUI())
                .subscribe {
                    pledge_details_label.text = getString(R.string.pledge_details_creator)
                    ViewUtils.setGone(received_section, true)
                    ViewUtils.setGone(delivery_disclaimer_section, it)
                }`

[the remark this answers is not in the document]
Comment (Arkariang) [reply app/src/main/java/com/kickstarter/ui/fragments/BackingFragment.kt:215]: 👍  perfect go ahead :)

**Comment by Arkariang** [comment]
Great work!

**Pull request merged**